modules/extension/ysld/src/main/java/org/geotools/ysld/parse/Util.java

```diff
@@ -218,7 +218,13 @@ public static Expression color(Object value, Factory factory) {
         if (value != null)
             value = value.toString();
 
-        return color != null ? factory.filter.literal(color) : expression((String) value, factory);
+        String hex = null;
+        if (color != null) {
+            hex = String.format("#%02x%02x%02x", color.getRed(), color.getGreen(), color.getBlue());
+            hex = hex.toUpperCase();
+        }
```

Comment:
Okay what is this logic trying to do? Is there a bug report?
 
There is a bunch of color converters that cover these cases, would be good to avoid duplication.